File: caption.py
import json
import time
import redis

# Adjust host/port if your Redis server uses different values
r = redis.Redis(host="127.0.0.1", port=6379)

meeting_id = "8f9a5dd0bdae1033a275102c12a7040b22a3ef22-1753867571123"
user_id = "w_1epve9s5bfus"
locale = "en-US"
caption_text = "Hello from a Python script!"

start_ms = int(time.time() * 1000)       # example timestamp
end_ms = start_ms + 2000                 # example end time

message = {
    "envelope": {
        "name": "UpdateTranscriptPubMsg",
        "routing": {"meetingId": meeting_id, "userId": user_id},
        "timestamp": int(time.time() * 1000),
    },
    "core": {
        "header": {
            "name": "UpdateTranscriptPubMsg",
            "meetingId": meeting_id,
            "userId": user_id,
        },
        "body": {
            "transcriptId": f"{user_id}-{start_ms}",
            "start": str(start_ms),
            "end": str(end_ms),
            "text": caption_text,
            "transcript": caption_text,
            "locale": locale,
            "result": True,
        },
    },
}

channel = "to-akka-apps-redis-channel"
r.publish(channel, json.dumps(message))

File: caption.js
const BigBlueButtonGW = require('./lib/bbb-gw');
const C = require('./lib/Constants');

const bbbGW = new BigBlueButtonGW();

function sendCaption(meetingId, userId, text, locale = 'en-US') {
  const payload = {
    envelope: {
      name: 'UpdateTranscriptPubMsg',
      routing: { meetingId, userId },
      timestamp: Date.now(),
    },
    core: {
      header: {
        name: 'UpdateTranscriptPubMsg',
        meetingId,
        userId,
      },
      body: {
        transcriptId: userId + '-0',
        start: '0',
        end: '0',
        text: '',
        transcript: text,
        locale,
        result: true,
      },
    },
  };

  // Publish to the same Redis channel as the controller
  bbbGW.publish(JSON.stringify(payload), C.TO_AKKA_APPS_CHAN_2x);
}

// Example usage
sendCaption('8f9a5dd0bdae1033a275102c12a7040b22a3ef22-1753909104959', 'w_5bt9yyqttlbb', 'Hello World');
